app/views/handlers/relationship_collector.py

```diff
@@ -118,3 +111,27 @@ def _is_last_relationship(self):
         if self.relationship_router.get_next_location_url(self._current_location):
             return False
         return True
+
+    def _render_block(self, block_id):
+        block_schema = self._schema.get_block(block_id)
+
+        transformed_block = transform_variants(
+            block_schema,
+            self._schema,
+            self._questionnaire_store.metadata,
+            self._questionnaire_store.answer_store,
+            self._questionnaire_store.list_store,
+            self._current_location,
+        )
+
+        placeholder_renderer = PlaceholderRenderer(
+            language=self._language,
+            schema=self._schema,
+            answer_store=self._questionnaire_store.answer_store,
+            metadata=self._questionnaire_store.metadata,
+            location=self._current_location,
+        )
```

Comment:
I think if you add location to the placeholder renderer initialisation in the base block handler you won't need to override the `_render_block` method.

Reply:
> I can't see any unit tests for location in placeholder parser. We also seem to be missing tests for list item id there as well weary

Will add these in now; just waiting to see if this is the preferred approach

Reply:
We already have this in v3 if you merge in the latest changes.